### PR TITLE
Convert the kyuubi batch conf with `spark.` prefix so that spark could identify

### PR DESCRIPTION
**Pull request by turboFei**
### _Why are the changes needed?_

We might need transfer some parameters like `kyuubi.client.ipAddress`, but it can not be identified by spark.

So we need convert them.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by turboFei** [comment]
cc @ulysses-you 
Sorry for the mistake in #3055, it can be identified by spark.

File: kyuubi-server/src/main/scala/org/apache/kyuubi/engine/spark/SparkBatchProcessBuilder.scala [hunk 1]
```diff
@@ -50,8 +50,16 @@ class SparkBatchProcessBuilder(
     KyuubiApplicationManager.tagApplication(batchId, "spark", clusterManager(), batchKyuubiConf)
 
     (batchKyuubiConf.getAll ++ sparkAppNameConf()).foreach { case (k, v) =>
+      val newKey = {
+        // convert the kyuubi config with `spark` prefix, so that spark could identify
+        if (k.startsWith("kyuubi.")) {
+          "spark." + k
```
Comment (ulysses-you) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/engine/spark/SparkBatchProcessBuilder.scala:56]: shall we reuse the code in `SparkProcessBuilder` ? so we can keep the same behavior on spark conf and hadoop conf.

Comment (turboFei) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/engine/spark/SparkBatchProcessBuilder.scala]: sorry for the late reply, updated.

**Review by ulysses-you — APPROVED** [review]

**Comment by turboFei** [comment]
thanks, merged to master

**Pull request closed without merging**